### DISCUSSION TITLE
yarn:outdated lists nothing

### DISCUSSION THREAD

**Discussion by jon-price-snke** [nickjj/docker-rails-example#35] [Q&A] [answered]
Hello, first of all...great work, this is awesome!

Everything works great, but I noticed that when running `./run yarn:outdated`, it returns nothing. I expected it to return outdated packages. For example, in this repo tailwind is running v3.0.7, but the latest version currently is v3.0.13. If I run `yarn outdated` it lists the outdated packages like I would expect it to. I've attached screenshots to show what I mean. Thanks!

![run yarn outdated](https://user-images.githubusercontent.com/63308415/149191154-f6b44198-2101-4b53-8afd-6bd2290f6289.png)
![yarn outdated](https://user-images.githubusercontent.com/63308415/149191151-8ed44f19-1797-46ca-ba18-9259c5f4ccd8.png)

**Comment by nickjj** [comment, marked as answer]
Hi,

Did you run `docker-compose up` in a different terminal before running `./run yarn:outdated`?

A lot of the run commands will connect to an already running container to run the command, if that container isn't up it silently fails. Although it sounds like this should throw an error and print out that the project isn't running.

**Reply by jon-price-snke** [reply]
Ah, I should've realized that. Sorry for wasting your time. It works fine with the container running. It would be nice to have the error just for dum dum's like me and probably wouldn't hurt to add `yarn:upgrade` command as well. Thanks!

**Reply by nickjj** [reply]
You're not dumb, this message should be printed for sure. I'm surprised it doesn't error out, something must be soaking up the error code because technically the container shouldn't be found. I'll for sure add it. Thanks for the report.

**Reply by jon-price-snke** [reply]
Awesome! One more question, what's the best way for adding new yarn packages? In the readme, it sounds like just adding the package:version into package.json is the way to do it. But the traditional way of `yarn add @fortawesome/fontawesome-free` doesn't work and errors out as seen in the screenshot below, since `node_modules` is being created within the dockerfile.

![yarn add package](https://user-images.githubusercontent.com/63308415/149213160-e1c237e2-b46b-4de0-986f-d3ba769b55fa.png)

**Reply by jon-price-snke** [reply]
It works if I do `docker compose run js yarn add @fortawesome/fontawesome-free`, but just want to check if that's the best way or if I'm missing something?

**Reply by nickjj** [reply]
Hi, yep you can do `docker compose run js yarn add @fortawesome/fontawesome-free`, although normally I go with your first method by adding the package to my `package.json` file by hand and then run `./run yarn:install`.

In general any command you find on the internet that says to run a Rails command or a Yarn command is exactly the same except you you need to prefix things with the Docker command first. For the Rails case there's a shortcut in the run script to be able to run `./run rails [ANY_COMMAND_YOU_WANT]`.

**Reply by jon-price-snke** [reply]
Ok cool, just wanted to check I was doing it right. The run file is very helpful indeed. Thanks!

**Reply by nickjj** [reply]
By the way, `docker-compose exec` doesn't show an error message when a container doesn't exist. This is the command that gets run when you run most `run` commands to do something in a container. This is an inherit problem with `docker-compose`. I could write additional logic in the script to handle this but it could lead to incorrect messages being shown if a different type of error happens that's not related to the container missing.
